### PR TITLE
adding IP Protocol enumeration to CRD can reduce the kube-ovn Controller judgment logic  

### PR DESCRIPTION
**Pull request by wangyd1988**
#### What type of this PR
Examples of user facing changes:
- API changes
- Bug fixes

#### Which issue(s) this PR fixes:
Fixes #(issue-number)

### CODE REVIEW

File: pkg/controller/subnet.go [hunk 1]
```diff
@@ -289,7 +289,7 @@ func formatSubnet(subnet *kubeovnv1.Subnet, c *Controller) error {
 		subnet.Spec.Provider = util.OvnProvider
 		changed = true
 	}
-	if subnet.Spec.Protocol == "" || subnet.Spec.Protocol != util.CheckProtocol(subnet.Spec.CIDRBlock) {
```
Comment (hongzhen-ma) [inline pkg/controller/subnet.go:292]: The protocol field can be filled automatically if it is not set. The subnet crd should be created as easy as posible.

Comment (wangyd1988) [reply pkg/controller/subnet.go:292]: if  i  create or update a error  ip  protocol(like  aaa )， subnet  crd help me changeing it to correct，i think it's not good idea. 
  If ip protocol filed is only be to demonstration  when someone  passes the  command  "kubectl get subnet "?  
I think it should be in the status  zone.

Comment (hongzhen-ma) [reply pkg/controller/subnet.go:292]: Are there some situations in your environment to set the protocol field(correct or wrong)，and is used specially ?

Comment (wangyd1988) [reply pkg/controller/subnet.go:292]: yes,  We support creating subnets using YAML. ApiServer  need to check whether the protocol fields are correct.

**Comment by hongzhen-ma** [comment]
欢迎来参加kube-ovn的双周会，或者在微信群里讨论下，你那环境上对subnet使用的一些建议

**Comment by oilbeater** [comment]
It LGTM, @wangyd1988 please rebase the master and fix the conflict

**Pull request closed without merging**